### PR TITLE
[FIX] purchase: vendor price not converted to default UoM

### PR DESCRIPTION
**Pull request by Florimond**
-Define "g" as the default UoM for a product.
-Make a purchase order with that product.
-Select the "kg" UoM in the order line and a price of 50.
-Check the Purchase tab of the product.

Before this commit:

the UoM of the price created is the default UoM for the product, "g".
This is because it's a related field.
However, the price wasn't converted to the default UoM, it's still 50.

After this commit:

the price is converted to the default UoM, it's 0.05 per "g".

OPW: 2076722

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Florimond** [comment]
@nim-odoo

File: addons/purchase/models/purchase.py [hunk 1]
```diff
@@ -450,13 +450,19 @@ def _add_supplier_to_product(self):
             # Do not add a contact as a supplier
             partner = self.partner_id if not self.partner_id.parent_id else self.partner_id.parent_id
             if partner not in line.product_id.seller_ids.mapped('name') and len(line.product_id.seller_ids) <= 10:
+                # Convert the price in the right currency
                 currency = partner.property_purchase_currency_id or self.env.user.company_id.currency_id
+                price = self.currency_id.compute(line.price_unit, currency, round=False)
+                # Compute the price for the template's UoM, because the supplier's UoM is related to that UoM.
+                if line.product_id.product_tmpl_id:
```
Comment (nim-odoo) [inline addons/purchase/models/purchase.py]: I'd rather do:
`if line.product_id.product_tmpl_id.uom_po_id != line.product_uom:`

**Comment by nim-odoo** [comment]
robodoo retry

**Comment by nim-odoo** [comment]
robodoo retry

**Comment by nim-odoo** [comment]
robodoo r+

@d-fence there is more and more often this error:
`AssertionError: ('Error loading file https://js.stripe.com/v3/\n`

**Comment by nim-odoo** [comment]
robodoo retry

**Comment by d-fence** [comment]
> robodoo r+
> 
> @d-fence there is more and more often this error:
> `AssertionError: ('Error loading file https://js.stripe.com/v3/\n`

It broke 12 times on 11.0.
It seems that it was bring to 11.0 with this 7264923b64e7bccb5c51bbbbd81f2e854bbbfa58

@dbo-odoo It's seems that an external library is needed in a non external test.
I don't know if the external code can be included in odoo or if the code is really needed by the website_event_sale.tests.test_ui.TestUI test, as it's not payment test ?

**Comment by dbo-odoo** [comment]
@d-fence 
indeed currently not used in any testing, but it's weird that it fails, stripe servers are very reliable (and they should be). Does headless chrome have a very small timeout value or something? Or is internet access limited for runbot containers?

I'm not sure if there's an easy way to detect from the JS if we are in a test suite or not, so not sure how easy it is to make this conditional (but I'd rather not anyway). Do you have any idea of how we could avoid this during testing?

**Pull request closed without merging**